src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs

```diff
@@ -2284,6 +2284,70 @@ public void TestWithModifiers_AllowedModifiers()
+            // Yes, it's correct that static constructors cannot have accessibility, but why not static is "replaced" with public?
+            // The same issue exists for file types that cannot have accessibility. Should attempting to make them public do "nothing"?
+            VerifySyntax<ConstructorDeclarationSyntax>(publicCtor, @"static C()
+{
+}");
+        }
+
+        [Fact]
+        public void TestAddAbstractToFileClass()
+        {
+            var fileClass = (ClassDeclarationSyntax)SyntaxFactory.ParseMemberDeclaration("file class C { }");
+            var fileAbstractClass = Generator.WithModifiers(fileClass, Generator.GetModifiers(fileClass).WithIsAbstract(true));
+            VerifySyntax<ClassDeclarationSyntax>(fileAbstractClass, @"file abstract class C
+{
+}");
+        }
+
+        [Fact]
+        public void TestAddPublicToFileClass()
+        {
+            var fileClass = (ClassDeclarationSyntax)SyntaxFactory.ParseMemberDeclaration("file class C { }");
+            var filePublicClass = Generator.WithAccessibility(fileClass, Accessibility.Public);
+            // bad output. It will be fixed in a later commit.
+            // This commit just demonstrates the *current* behavior (test is passing currently, while it shouldn't).
+            // What 's the expected behavior is unclear given an existing behavior for static constructors demonstrated in this commit (the new test is passing)
+            // The current behavior might be correct if we'll follow the static constructors behavior.
+            VerifySyntax<ClassDeclarationSyntax>(filePublicClass, @"file class C
+{
+}");
+        }
+
+        [Fact]
+        public void TestAddFileModifierToAbstractClass()
+        {
+            var abstractClass = (ClassDeclarationSyntax)SyntaxFactory.ParseMemberDeclaration("abstract class C { }");
+            var fileAbstractClass = Generator.WithModifiers(abstractClass, Generator.GetModifiers(abstractClass).WithIsFile(true));
+            VerifySyntax<ClassDeclarationSyntax>(fileAbstractClass, @"file abstract class C
+{
+}");
+        }
+
+        [Fact]
+        public void TestAddFileModifierToPublicClass()
+        {
+            var publicClass = (ClassDeclarationSyntax)SyntaxFactory.ParseMemberDeclaration("public class C { }");
+            var filePublicClass = Generator.WithModifiers(publicClass, Generator.GetModifiers(publicClass).WithIsFile(true));
+            // bad output. It will be fixed in a later commit.
+            // This commit just demonstrates the *current* behavior (test is passing currently, while it shouldn't).
+            // note, behavior changed here in this commit since we supported file, but the most correct behavior is to eliminate public completely.
+            // note 2, the correct behavior here is actually not very clear given the constructor case and the opposite case of this.
+            VerifySyntax<ClassDeclarationSyntax>(filePublicClass, @"public file class C
+{
+}");
+        }
```

Comment:
This is even more confusing for a case where the class originally didn't have any accessibility or modifiers (ie, a plain `class C`), and `WithModifiers` is called with `DeclarationModifiers.Public | DeclarationModifiers.File`. Yes it's weird that someone do this, but.. it's a public API.

Reply:
i'm ok with this just throwing away public modifiers in that case.  we did similar things with interface members (prior to them supporting accessibility or method bodies).

Reply:
This is actually my bad. There is no `DeclarationModifiers.Public` :smile:

The way to say both `public` and `file` will be two calls, one to `WithModifiers` and one to `WithAccessibility`. The second call will overwrite the first in this case.